### ISSUE TITLE
Storybook Accessibility Violations

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#6124]
Recently the addon-a11y tool was added to storybook which has flagged a number of accessibility violations. To see the violations go to [storybook](https://storybook.grommet.io/?path=/story/all--all) and below each story click the accessibility tab. It would be great to clean up these accessibility violations. Some fixes may just require changes to story while others will require changes to grommet components. Below is a list of stories with accessibility violations. Feel free to contribute fixes per story or per component.

- [ ] All (need design help - Calendar color contrast & Calendar heading level https://github.com/grommet/grommet/issues/7450)

**Accordion**
- [x] Header https://github.com/grommet/grommet/pull/6383
- [x] CustomThemed/Rich

**Drop**
- [x] https://github.com/grommet/grommet/pull/6384
- (Inline, Overflow, Plain, Simple, Styled)

**Menu**
- [x] Children #6410 

**Pagination**

- [x] (component wide error) `<ul>` and `<ol>` must only directly contain `<li>`, `<script>` or `<template>` elements
- [x] Number edge pages
- [x] Number middle pages
- [x] Simple
- [x] Size

**CheckBox**
- [x] 	Inside a Button
- [x] No label https://github.com/grommet/grommet/pull/6449
	
**FileInput** 
- [x] https://github.com/grommet/grommet/pull/6417
	
**Form**
- [x] Aggreagate validaton https://github.com/grommet/grommet/pull/6469
- [x] Array of Form Fields https://github.com/grommet/grommet/pull/6464
- [x] Controlled
- [x] Controlled Input
- [x] Controlled Input lazy
- [x] Controlled lazy
- [x] Dynamic fields https://github.com/grommet/grommet/pull/6465
- [ ] Field states (need design help)
- [x] Field with children
- [x] Field with component prop
- [ ] Help and error (need design help)
- [x] Typed form
- [x] Uncontrolled
- [x] Validate on blur https://github.com/grommet/grommet/pull/6466
- [x] Validate on change
- [ ] Validate on mount (need design help)
- [x] Custom
- [ ] Field spacing options (need design help)
- [ ] Required label (need design help)
- [x] Themed

**FormField**
- [x] Simple
- [x] Custom

**MaskedInput**
- [x] https://github.com/grommet/grommet/pull/6448

**RangeInput**
- [x] https://github.com/grommet/grommet/pull/6423

**Select**
- [x] (component wide error) Interactive controls must not be nested

**TextArea**
- [x] https://github.com/grommet/grommet/pull/6374
- [x] Themed

**TextInput**
- [x] https://github.com/grommet/grommet/pull/6425
- [x] Simple
	
**Box**
- [x] Background
- [x] Fixed sizes https://github.com/grommet/grommet/pull/6459
- [x] Min and max sizes https://github.com/grommet/grommet/pull/6460
- [x] Round
- [x] Simple

**Card**
- [x] Simple
- [x] Clickable
- [x] Stacked
- [x] Themed

**Footer**
- [x] Double footer
- [x] Sitemap
- [x] Social

**Header**
- [x] https://github.com/grommet/grommet/pull/6419
(Responsive, Simple)

**Layer**
- [x] Target

**Sidebar**
- [x] Labels https://github.com/grommet/grommet/pull/6373

**Carousel**

- [x] Multi #6414 

**Markdown**
- [x] Component override markdown https://github.com/grommet/grommet/pull/6453

**Paragraph**

- [ ] All (need design help)

**AnnounceContext**
- [x] https://github.com/grommet/grommet/pull/6442

**Grommet**
- [x] Messages https://github.com/grommet/grommet/pull/6442
- [x] Vars

**InfiniteScroll**
- [x] https://github.com/grommet/grommet/pull/6420
(Grid, Grid with show item 77)

**SkipLinks**
- [x] Simple https://github.com/grommet/grommet/pull/6446

**Theme**

- [x] Colors

**ThemeContext**

- [x] Extend

**Chart**
- [x] https://github.com/grommet/grommet/pull/6422

**DataTable**

- [x] Controlled https://github.com/grommet/grommet/pull/6440
- [x] rowDetails https://github.com/grommet/grommet/pull/6441
- [ ] OnUpdate (need design help)
- [x] Paginated
- [x] Styled https://github.com/grommet/grommet/pull/6454
- [x] Sized
- [x] Fill and pin

**Distribution**

- [x] Simple https://github.com/grommet/grommet/pull/6510

**List**

- [x] Selection
- [x] Order
- [x] onClickItem
- [x] Key Render
- [x] Paginated

**Notification**
- [x] https://github.com/grommet/grommet/pull/6429

**ToggleGroup**
- [x] Controlled
- [x] Multiple

**Data**
- [x] Cards
- [x] OnSelect
- [x] Properties

**DataSort**
- [x] Simple

**Button**
- [x] Kind
- [x] TS Custom

**Drop**
- [x] Lazy
- [x] Themed

**Nav**
- [x] Justify Sidebar

**Tabs**
- [x] Responsive
- [x] Custom

**DataSummary**
- [x] Simple

**CheckBox**
- [x] Custom
- [x] Custom toggle

**DateInput**
- [ ] Inline (need design help)

**Grid**
- [x] Area prop alternatives

**PageHeader**
- [x] Level
- [x] Size

**SkipLinks**
- [x] Themed

**Calendar**
- [ ] component wide contrast issue (need design help)
- [x] Daylight savings time
- [x] Show adjacent days

**Cards**
- [x] Children

**Skeleton**
- [x] GridContainer
- [x] Group
- [x] Individual
- [x] Simple
- [x] Custom

**Comment by julieg18** [comment]
Hello, I'd love to help improve the accessibility for the storybooks! Can you assign me?

**Comment by jcfilben** [comment]
> Hello, I'd love to help improve the accessibility for the storybooks! May I be assigned to this issue?

Hi @julieg18 great to hear you are interested in working on this! I think since this is a larger issues items on the list can be handled either by component or one by one through separate pull requests. There may be many contributors to this issue so I am going to hold off assigning any one person to this, but feel free to start filing pull requests against this issue. Let me know if there are any questions!

**Comment by UmairAbbas78** [comment]
Hi, I would like to fix the contrast issues from the footer elements.
I can't access the footer elements as the initial path is Grommet testing page. Can someone please guide me through this.

**Comment by jcfilben** [comment]
Closing the remaining issues need some design adjustments and are tracked in separate tickets:

https://github.com/grommet/grommet-theme-hpe/issues/416
https://github.com/grommet/grommet/issues/7440
https://github.com/grommet/grommet/issues/7423
